packages/shared/src/utils/url/url.js

```diff
@@ -18,13 +18,17 @@ export const legacyUrlForLanguage = (target_language, url = window.location.href
     url.replace(new RegExp(`/${default_language}/`, 'i'), `/${(target_language || 'EN').trim().toLowerCase()}/`);
 
 export const urlForLanguage = (lang, url = window.location.href) => {
+    const current_url = new URL(url);
+    const params = new URLSearchParams(current_url.search.slice(1));
+
+    if (lang === 'EN') {
+        return `${current_url.origin}${current_url.pathname}`;
```

Comment:
why do we ignore `${params.toString()}${current_url.hash}`(it is in line 34)  if the lang is `EN`?

Reply:
We don't want to show english language in our query string, so we return url without it when language is english

Reply:
I refactored the language url function, so it should not affect other query parameters in any case